src/lib/verifier/prod.ml

```diff
@@ -157,20 +143,23 @@ module Worker_state = struct
                [%log internal] "Verifier_verify_commands_done" ;
                result
 
-             let verify_blockchain_snarks = B.Proof.verify
-
              let verify_blockchain_snarks bs =
                Context_logger.with_logger (Some logger)
                @@ fun () ->
                Internal_tracing.Context_call.with_call_id
                @@ fun () ->
                [%log internal] "Verifier_verify_blockchain_snarks" ;
-               let%map result = verify_blockchain_snarks bs in
+               let%map result =
+                 Blockchain_snark_state.verify ~key:verification_key bs
+               in
                [%log internal] "Verifier_verify_blockchain_snarks_done" ;
                result
 
              let verify_transaction_snarks ts =
-               match Or_error.try_with (fun () -> T.verify ts) with
+               match
+                 Or_error.try_with (fun () ->
+                     Transaction_snark.verify ts ~key:verification_key )
```

Comment:
This is incorrect. When we verify a transaction snark proof, we need to use the transaction snark verification key, not the blockchain snark verification key. You'll need to pipe that through.